src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -182,6 +185,25 @@ namespace OpenMS
     template<typename T>
     ExitCodes run(FASTAContainer<T>& proteins, std::vector<ProteinIdentification>& prot_ids, std::vector<PeptideIdentification>& pep_ids)
     {
+      // no decoy string provided? try to deduce from data
+      if (decoy_string_.empty())
+      {
+        LOG_INFO << "No decoy string provided. Trying to determine from data." << std::endl;
+        ExitCodes e = findDecoyString_(proteins, decoy_string_, prefix_);
```

Comment:
why return `ExitCodes`? there are only two outcomes: found, not-found, so a bool would be much more intuitive.